kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java

```diff
@@ -0,0 +1,323 @@
+                                                           "Solver '" + solverId + "' not found in container '" + containerId + "'",
+                                                           null );
+            }
+        } catch (Exception e) {
+            logger.error("Error retrieving solver '" + solverId + "' state from container '" + containerId + "'", e);
+            return new ServiceResponse<Solution>(ServiceResponse.ResponseType.FAILURE,
+                                                       "Error retrieving solver '" + solverId + "' state from container '" + containerId + "'" + e.getMessage(),
+                                                       null );
+        }
+    }
+
+    public ServiceResponse<SolverInstance> updateSolverState( String containerId, String solverId, SolverInstance instance ) {
+        try {
+            if( instance.getStatus() == null ) {
+                return new ServiceResponse<SolverInstance>(ServiceResponse.ResponseType.FAILURE,
+                                                           "Solver status is a mandatory field on an update call.",
+                                                           instance );
+            }
+            if( instance.getStatus() != SolverInstance.SolverStatus.NOT_SOLVING && instance.getStatus() != SolverInstance.SolverStatus.SOLVING ) {
+                return new ServiceResponse<SolverInstance>(ServiceResponse.ResponseType.FAILURE,
+                                                           "Invalid solver status. Only SOLVING or NOT_SOLVING status can be set.",
+                                                           instance );
+            }
+            SolverInstanceContext sic = solvers.get( SolverInstance.getSolverInstanceKey( containerId, solverId ) );
+            if( sic != null ) {
+                synchronized ( sic ) {
+                    if( sic.getInstance().getStatus() == SolverInstance.SolverStatus.NOT_SOLVING ) {
+                        if( instance.getStatus() == SolverInstance.SolverStatus.SOLVING ) {
+                            if( instance.getPlanningProblem() == null ) {
+                                return new ServiceResponse<SolverInstance>(ServiceResponse.ResponseType.FAILURE,
+                                                                           "Planning-problem is a mandatory field when starting the solver.",
+                                                                           instance );
+                            }
+                            startSolver( sic, instance );
+                        } else if( instance.getStatus() == SolverInstance.SolverStatus.NOT_SOLVING ) {
+                            return new ServiceResponse<SolverInstance>(ServiceResponse.ResponseType.SUCCESS,
+                                                                       "Solver '" + solverId + "' on container '" + containerId + "' is already executing.",
```

Comment:
It seems that this response text should be switched with response text on line 199. Here it is returned in case original status is NOT_SOLVING and updated status is NOT_SOLVING too. So response here should be that solving didn't started yet or already terminated.